pkg/statedb/index/keyset.go

```diff
@@ -5,12 +5,19 @@ package index
 
 import "bytes"
 
-type Key = []byte
+// Key is a byte slice describing a key used in an index by statedb.
+// If a key is variable-sized, then it must be either terminated with
+// e.g. zero byte or it must be length-encoded. If it is not, then
+// Get() on a unique index may return unmatching results.
```

Comment:
I don't think this applies only to unique indices, as this comment suggests. I initially found the issue on a non-unique index (device name in `Table[NodeAddress]`).

Reply:
Good point, reworded.